example/example/autoextract.py

```diff
@@ -19,16 +19,13 @@ class AutoextractProductResponse:
     data: Dict[str, Any]
 
 
-@provides(AutoextractProductResponse)
 class AutoextractProductProvider(PageObjectInputProvider):
-
-    def __init__(self, response: DummyResponse):
-        self.response = response
+    provided_classes = {AutoextractProductResponse}
 
     @inlineCallbacks
-    def __call__(self):
-        data = (yield get_autoextract_product(self.response.url))
-        res = AutoextractProductResponse(data=data)
+    def __call__(self, to_provide, request: Request):
+        data = (yield get_autoextract_product(request.url))
+        res = {AutoextractProductResponse: AutoextractProductResponse(data=data)}
         raise returnValue(res)
```

Comment:
returnValue shouldn't be needed here, I recall that was only needed for Python2 compatibility

```suggestion
        return {AutoextractProductResponse: AutoextractProductResponse(data=data)}
```